### PR TITLE
In-line wasm ready check for schnorrkel signing

### PR DESCRIPTION
**Pull request by jordy25519**

### CODE REVIEW

File: src/doughnut/signingMethods/0/index.js [hunk 1]
```diff
@@ -5,9 +5,23 @@ const {
...
+
+function wait_ms(ms) {
+  var start = Date.now();
+  while (Date.now() - start < ms) {}
+}
+
+// Wrap schnorrkel signing to await wasm crypto readiness
+function sign(payload, keypair) {
+    for(i = 0; i < 10; i++) {
+      if(!isReady()) wait_ms(10);
```
Comment (ianhe8x) [inline src/doughnut/signingMethods/0/index.js]: blocking-style wait sounds a bad idea.

Comment (jordy25519) [reply src/doughnut/signingMethods/0/index.js]: I was hoping you could "hide" as async function but in JS but you can't

File: src/doughnut/signingMethods/1/index.js [hunk 2]
```diff
@@ -6,8 +6,12 @@ const {
   naclVerify,
 } = require("@polkadot/util-crypto");
 
+async function sign(payload, keypair) {
```
Comment (xlc) [inline src/doughnut/signingMethods/1/index.js]: this change doesn't make any real differences

File: src/doughnut/signingMethods/0/index.js (hunk 1 above)
Comment (wilfred-centrality) [reply src/doughnut/signingMethods/0/index.js]: @holygits let's have `wait_ms` use a `setTimeout` and return a promise. That will make it `await`able, and not block.

File: src/doughnut/signingMethods/1/index.js (hunk 2 above)
Comment (ianhe8x) [reply src/doughnut/signingMethods/1/index.js]: it's for keep the interface same with schnorrkel. `async` makes it returns Promise

File: src/doughnut/signingMethods/0/index.js [hunk 3]
```diff
@@ -2,12 +2,19 @@
...
+// Wrap schnorrkel signing to await wasm crypto readiness
+async function sign(payload, keypair) {
+    await cryptoWaitReady();
+    return schnorrkelSign(payload, keypair);
+}
+
 module.exports = {
-  sign: schnorrkelSign,
+  sign: sign,
   verify: schnorrkelVerify,
```
Comment (wilfred-centrality) [inline src/doughnut/signingMethods/0/index.js]: Does schnorrkel verify not require the same treatment? (same question for signing method 1)

**Review by wilfred-centrality — CHANGES_REQUESTED** [review]

File: src/doughnut/signingMethods/0/index.js (hunk 3 above)
Comment (jordy25519) [reply src/doughnut/signingMethods/0/index.js]: Yes, very good point! updated.

**Review by wilfred-centrality — APPROVED** [review]

**Pull request merged**